### DISCUSSION TITLE
Reset form with react-text-mask seems to unregister the input

### DISCUSSION THREAD

**Discussion by jpc0rrea** [react-hook-form/react-hook-form#5221] [Q&A] [answered]
**Describe the bug**
While using `react-text-mask`, on the first time I fill the form everything goes normal. After I call de `reset` function, from the `useForm` hook, the masked input value seems to be "unregistered" from the form, because its value stops being updated. The value from the input with mask it "stuck" on the value that I reseted.

**To Reproduce**

1. Fill the first input;
2. Fill the second input with a number;
3. Press "Create transaction" and wait;
4. After the first transaction is created, fill the first input again;
5. Fill the second input;
6. See the error;

**Codesandbox link (Required)**
[Codesandbox link](https://codesandbox.io/s/serene-dew-w68qh?file=/src/App.js).

**Expected behavior**
After I reset the form, the form continues to validate properly my masked input and update it's value as I type;

**Screenshots**
Using the reset, with the error:
![Using the reset, with the error](https://media.giphy.com/media/Q9RWjll3GaRMslDUGF/giphy.gif)

Without reset, without the error:
![Without reset, without the error](https://media.giphy.com/media/kppIel9RB32e8MAxXs/giphy.gif)

**Desktop (please complete the following information):**

- OS: macOS Big Sug 11.3.1
- Browser: Safari
- Version: 14.1

**Additional context**
I'm using a custom function with `yup` to validate the masked input information. I'm also using `chakra-ui` to create the components.

**Comment by leapful** [comment, marked as answer]
@jpc0rrea 

This is not a bug. By default, react-text-mask forwards a custom ref object which is not a DOM reference of input element for "ref" callback, therefore it will be not compatible with react-hook-form ref handling.

Just set the correct DOM element from prop "inputElement" of react-text-mask custom ref object inside MaskedInput ref handling will fix the issue.

Working sample: https://codesandbox.io/s/dark-frog-h5vdv?file=/src/Components/MaskedInput.js

**Reply by bluebill1049** [reply]
Thank you @leapful 🙏

**Comment by jpc0rrea** [comment]
@leapful Thanks a lot! It resolved the problem for me! Can you help me with the types?

I typed my `MaskedInputBase` as `ForwardRefRenderFunction<HTMLInputElement,MaskedInputProps>`, and my `MaskedInputBase` as:

`
interface MaskedInputProps extends ChakraInputProps {
  name: string;
  mask: string;
  label?: string;
  error?: FieldError;
}
`

But it's giving me the error "Property 'inputElement' does not exist on type 'HTMLInputElement'", because the `data` argument from the ref callback is typed as `HTMLInputElement`. Any thoughts on this?

**Reply by leapful** [reply]
@jpc0rrea 

Please share your CSB with Typescript. Your current CSB is using javascript.

**Reply by jpc0rrea** [reply]
@leapful 

The link for the CSB with Typescript is [here](https://codesandbox.io/s/serene-dew-w68qh?file=/src/Components/MaskedInput.tsx)

**Reply by leapful** [reply]
@jpc0rrea 

You can override the type definition of prop "ref" in ChakraInputProps when re-assign the component.

For example:
```
const MaskedChakraInput: ComponentWithAs<'input', MaskedChakraInputProps> = ChakraInput
```

Working sample: https://codesandbox.io/s/heuristic-resonance-ptxll?file=/src/Components/MaskedInput.tsx

**Reply by jpc0rrea** [reply]
Thanks @leapful! It's working perfectly!

**Reply by bluebill1049** [reply]
thanks @leapful
